extensions/Khronos/KHR_materials_common_v2/README.md

```diff
@@ -0,0 +1,250 @@
+The following example defines a Lambert shaded material with a diffuse texture,
+moderate shininess and red specular highlights. 
+
+```json
+    "materials": [
+        {
+            "name": "lambert_textured_material",
+            "emissiveTexture" : {
+                "index" : 0
+            },
+            "extensions": {
+                "KHR_materials_common": {
+                    "technique": "LAMBERT",
+                    "diffuseTexture" : {
+                        "index" : 1
+                    }
+                }
+            }
+        }
+    ]
+```
+</details>
+
+### Blinn-Phong
+
+The `BLINNPHONG` technique is intended for shading according to the Blinn-Phong
+lighting model or a close approximation. The underlying equation is detailed by
+ACM: “Models of Light Reflection for Computer Synthesized Pictures,” SIGGRAPH
+77, pp 192-198
+[http://portal.acm.org/citation.cfm?id=563893](http://portal.acm.org/citation.cfm?id=563893).
+In particular see the section, "Simple Highlight Model."
+
+The following code illustrates the computation:
+
+```
+color = emissiveTerm + occlusionTerm * (diffuseTerm * max(N * L, 0) + specularTerm * max(H * N, 0)^shineTerm)
+```
+
+> **Implementation Note**: Writers should be aware of the range of the specular
+> exponent (`shininess`), which is _not_ a normalized range. Concretely
+> speaking, given the above equation, a `shininess` value of 1.0 corresponds to
+> a very low shininess. For orientation: using the traditional OpenGL fixed
+> function pipeline, the specular exponent was expected to be within [0, 128].
+> However, using glTF, larger `shininess` values are clearly possible.
+
+> **TODO:** Mention caution with `shininess < 1`?
+
+<details>
+    <summary>Example: Blinn-Phong</summary>
+The following example defines a Blinn-Phong shaded material with a diffuse texture, moderate shininess and red specular highlights.
+
+```json
+    "materials": [
+        {
+            "name": "shiny_blinnphong_material",
+            "extensions": {
+                "KHR_materials_common": {
+                    "technique": "BLINNPHONG",
+                    "diffuseFactor" : [
+                        0.8, 
+                        0.8, 
+                        0.8, 
+                        1.0
+                    ], 
+                    "diffuseTexture" : {
+                        "index" : 0
+                    }, 
+                    "shininessFactor" : 10.0, 
+                    "specularFactor" : [
+                        1, 
+                        0, 
+                        0
+                    ]
+                }
+            }
+        }
+    ]
+```
+</details>
+
+## Extending Materials
+
+Materials are defined by adding the `KHR_materials_common` extension to any
+glTF material. When the extension is provided alongside a core glTF material
+or another material extension, client implementations may choose the most
+appropriate material for performance or other constraints.
+
```

Comment:
I have been wondering if we should attempt to guide the way in which clients render situations where multiple material extensions are present, or this extension along with pbrMetallicRoughness. From one point of view, a filled-out PBR struct would always be primary, and e.g. BLINNPHONG might be sent along as a fallback for weak devices. From another point of view, a LAMBERT material would be the model creator's ideal aesthetic, but they also fill out the PBR struct with approximate values for clients that don't accept the extension. Does each client guess? Do we guide the guesswork? Do we include a boolean flag or some-such in this extension to help suggest "this is a fallback" vs "this is the ideal rendering path"? (I think that's probably not a good idea, but who knows.)